homeassistant/components/screenlogic/__init__.py

```diff
@@ -124,8 +132,22 @@ async def _async_update_data(self):
         try:
             async with self.api_lock:
                 await self.hass.async_add_executor_job(self.gateway.update)
-        except ScreenLogicError as error:
-            raise UpdateFailed(error) from error
+        except ScreenLogicError:
+            _LOGGER.warning("ScreenLogicError - attempting reconnect")
+
+            try:
+                _LOGGER.warning("Re-connecting to the gateway")
+                connect_info = get_connect_info(self.hass, self.config_entry)
+                self.gateway = ScreenLogicGateway(**connect_info)
+                self.gateway.update()
+            except ScreenLogicError as error:
+                _LOGGER.error(
+                    "Error while re-connecting to the gateway %s: %s",
```

Comment:
Maybe we dont need to log that we failed during reconnection? The coordinator will already log the update failed error.